### PR TITLE
Use a negotiation options to wrap all the parameter

### PR DESCRIPTION
**Pull request by Y-Sindo**
Ensure test cover negotation options in null

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs [hunk 1]
```diff
@@ -48,7 +46,7 @@ public ServiceHubContext(string hubName, IHubContext<Hub> hubContext, IServiceHu
 
         Task<NegotiationResponse> IInternalServiceHubContext.NegotiateAsync(HttpContext httpContext, string userId, IList<Claim> claims, TimeSpan? lifetime, bool isDiagnosticClient, CancellationToken cancellationToken)
         {
-            return _negotiateProcessor.NegotiateAsync(_hubName, httpContext, userId, claims, lifetime, isDiagnosticClient, cancellationToken);
+            return _negotiateProcessor.NegotiateAsync(_hubName, options, cancellationToken);
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs:49]: no assignment for `options`

File: src/Microsoft.Azure.SignalR.Management/NegotiateProcessor.cs [hunk 2]
```diff
@@ -26,8 +26,17 @@ public NegotiateProcessor(IServiceEndpointManager serviceEndpointManager, IEndpo
             _router = router;
         }
 
-        public async Task<NegotiationResponse> NegotiateAsync(string hubName, HttpContext httpContext = null, string userId = null, IEnumerable<Claim> claims = null, TimeSpan? lifetime = null, bool isDiagnosticClient = false, CancellationToken cancellationToken = default)
+        public async Task<NegotiationResponse> NegotiateAsync(string hubName, NegotiationOptions negotiationOptions, CancellationToken cancellationToken = default)
         {
+            if (negotiationOptions == null)
+            {
+                negotiationOptions = new NegotiationOptions();
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/NegotiateProcessor.cs:33]: disallow `null` or create an internal `Default` instance.

File: src/Microsoft.Azure.SignalR.Management/IInternalServiceHubContext.cs [hunk 3]
```diff
@@ -16,14 +13,8 @@ internal interface IInternalServiceHubContext : IServiceHubContext
...
-        /// <param name="claims">The claim list to be put into access token. If null, the claims in <see cref="HttpContext.User"/> of <paramref name="httpContext"/> will be used.</param>
-        /// <param name="lifetime">The lifetime of the token. The default value is one hour.</param>
-        /// <param name="isDiagnosticClient">The flag whether the client to be connected is a diagnostic client.</param>
-        /// <param name="cancellationToken">Cancellation token for aborting the operation. If null, the <see cref="HttpContext.RequestAborted"/> of <paramref name="httpContext"/> will be used. </param>
         /// <returns>Client endpoint and access token to Azure SignalR Service.</returns>
-        Task<NegotiationResponse> NegotiateAsync(HttpContext httpContext = null, string userId = null, IList<Claim> claims = null, TimeSpan? lifetime = null, bool isDiagnosticClient = false, CancellationToken cancellationToken = default);
+        Task<NegotiationResponse> NegotiateAsync(NegotiationOptions options = null, CancellationToken cancellationToken = default);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/IInternalServiceHubContext.cs:17]: not necessary to allow `null` options, add a defense check is enough?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/IInternalServiceHubContext.cs:17]: https://github.com/Azure/azure-signalr/blob/b54a50f41486d4ca590f6ce2b8ecdc72cfee1501/src/Microsoft.Azure.SignalR.Management/ServiceManager.cs#L63 
`ServiceManager` allows `userId` to be null, so the scene with no parameter is valid.

**Review by vwxyzh — APPROVED** [review]

**Pull request merged**